apps/daimo-mobile/src/view/shared/AmountInput.tsx

```diff
@@ -74,16 +77,26 @@ function AmountInput({
   const change = useCallback((text: string) => {
     if (disabled) return;
 
-    setStrVal(text);
-    const newVal = parseLocalFloat(text);
+    // Validate. Handle negative numbers, NaN, out of range.
+    const looksValid = /^(|0|(0[.,]\d*)|([1-9]\d*[.,]?\d*))$/.test(text);
```

Comment:
took a while to get this right

tested with both the US `0.00` and UK `0,00` locale